src/video/kmsdrm/SDL_kmsdrmopengles.c

```diff
@@ -83,7 +98,263 @@ bool KMSDRM_GLES_SetSwapInterval(SDL_VideoDevice *_this, int interval)
     return true;
 }
 
-bool KMSDRM_GLES_SwapWindow(SDL_VideoDevice *_this, SDL_Window *window)
+static EGLSyncKHR create_fence(SDL_VideoDevice *_this, int fd)
+{
+    EGLint attrib_list[] = {
+        EGL_SYNC_NATIVE_FENCE_FD_ANDROID, fd,
+        EGL_NONE,
+    };
+
+    EGLSyncKHR fence = _this->egl_data->eglCreateSyncKHR(_this->egl_data->egl_display, EGL_SYNC_NATIVE_FENCE_ANDROID, attrib_list);
+
+    SDL_assert(fence);
+    return fence;
+}
+
+/***********************************************************************************/
+/* Comments about buffer access protection mechanism (=fences) are the ones boxed. */
+/* Also, DON'T remove the asserts: if a fence-related call fails, it's better that */
+/* program exits immediately, or we could leave KMS waiting for a failed/missing   */
+/* fence forevever.                                                                */
+/***********************************************************************************/
+static bool KMSDRM_GLES_SwapWindowFenced(SDL_VideoDevice *_this, SDL_Window * window)
+{
+    SDL_WindowData *windata = ((SDL_WindowData *) window->internal);
+    SDL_DisplayData *dispdata = SDL_GetDisplayDriverDataForWindow(window);
+    KMSDRM_FBInfo *fb;
+    KMSDRM_PlaneInfo info;
+    bool modesetting = false;
+
+    SDL_zero(info);
+
+    /******************************************************************/
+    /* Create the GPU-side FENCE OBJECT. It will be inserted into the */
+    /* GL CMDSTREAM exactly at the end of the gl commands that form a */
+    /* frame.(KMS will have to wait on it before doing a pageflip.)   */
+    /******************************************************************/
+    dispdata->gpu_fence = create_fence(_this, EGL_NO_NATIVE_FENCE_FD_ANDROID);
+    SDL_assert(dispdata->gpu_fence);
```

Comment:
Based on the preceding comment:
>Also, DON'T remove the asserts: if a fence-related call fails, it's better that program exits immediately, or we could leave KMS waiting for a failed/missing fence forever.

 this should be SDL_assert_always(), or the asserts will be compiled out in release builds.
```suggestion
    SDL_assert_always(dispdata->gpu_fence);
```